version: '2'

services:
  django: &django
    image: geekbeacon/django
    build:
      context: .
      dockerfile: ./compose/local/django/Dockerfile
    depends_on:
      - postgres
      - rabbit
    volumes:
      - .:/app
    environment:
    # correct database url fills out all fields for database settings
    # avoids problems with invalid DB settings running python manage.py tasks from shell
      - DATABASE_URL=postgres://geek_beacon:geek_beacon@postgres/geek_beacon
      - POSTGRES_USER=geek_beacon
      - POSTGRES_PASSWORD=geek_beacon
      - USE_DOCKER=yes
    ports:
      - "8000:8000"
    command: /start.sh
  postgres:
    image: geekbeacon/postgres
    build:
      context: .
      dockerfile: ./compose/production/postgres/Dockerfile
    environment:
      - POSTGRES_USER=geek_beacon
      - POSTGRES_PASSWORD=geek_beacon
    volumes:
      - ./volumes/postgres_data_local:/var/lib/postgresql/data
      - ./volumes/postgres_backup_local:/backups
    ports:
      - "5432:5432"
  memcached:
    image: memcached:latest
  rabbit:
    image: rabbitmq:latest
    environment:
      - RABBITMQ_DEFAULT_USER = admin
      - RABBITMQ_DEFAULT_PASS = mypass
  celeryworker:
    <<: *django
    depends_on:
      - rabbit
      - postgres
    ports: []
    command: /start-celeryworker.sh
  celerybeat:
    # https://github.com/docker/compose/issues/3220
    <<: *django
    depends_on:
      - rabbit
      - postgres
    ports: []
    command: /start-celerybeat.sh
